### PR TITLE
Merge the two plugin guides into one

### PR DESCRIPTION
**Pull request by alcuadrado**

### CODE REVIEW

File: docs/advanced/building-plugins.md [hunk 1]
```diff
@@ -1,13 +1,70 @@
...
+## What exactly are plugins in Hardhat?
 
-Plugins are bits of reusable configuration. Anything that you can do in a plugin, can also be done in your config file. You can test your ideas in a config file, and move them into a plugin when ready.
+Plugins are bits of reusable configuration. Anything that you can do in a plugin, can
+also be done in your config file. You can test your ideas in a config file, and move
+them into a plugin when ready.
 
-The main things that plugins can do are extending the Hardhat Runtime Environment, extending the Hardhat config, defining new tasks, and overriding existing ones.
+When developing a plugin the main tools available to integrate new functionality are extending the
+[Hardhat Runtime Environment], extending the Hardhat config, defining new tasks and
```
Comment (fvictorio) [inline docs/advanced/building-plugins.md:13]:
```suggestion
[Hardhat Runtime Environment](../advanced/hardhat-runtime-environment), extending the Hardhat config, defining new tasks and
```

File: docs/advanced/building-plugins.md [hunk 2]
```diff
@@ -1,13 +1,70 @@
...
+overriding existing ones, which are all configuration actions achieved through code.
+
+Some examples of things you could achieve by creating a plugin are running a linter when
+the `check` task runs, using different compiler versions for different files or
+generating an UML diagram for your contracts.
+
+
+## Extending the Hardhat Runtime Environment
+
+Let’s go through the process of creating a plugin to inject ethers.js to the [Hardhat Runtime Environment].
```
Comment (fvictorio) [inline docs/advanced/building-plugins.md:23]:
```suggestion
Let’s go through the process of creating a plugin to inject ethers.js to the Hardhat Runtime Environment.
```

File: docs/advanced/building-plugins.md [hunk 3]
```diff
@@ -1,13 +1,70 @@
...
+```
+
+This is literally all it takes to put together a plugin for Hardhat. Now `hi` is available to be used in 
+the Hardhat console, your tasks, tests and other plugins. 
+
+## Using the Hardhat TypeScript plugin boilerplate
+
+For a complete example of a plugin you can take a look at the [Hardhat TypeScript plugin boilerplate project](https://github.com/nomiclabs/hardhat-ts-plugin-boilerplate/).
+
+Plugins don't need to be written in TypeScript, but we recommend doing it, as many of our use it. Creating a plugin in
```
Comment (fvictorio) [inline docs/advanced/building-plugins.md:66]: "many of our use it"? Many of our users?

File: docs/advanced/building-plugins.md [hunk 4]
```diff
@@ -17,19 +74,26 @@ Make sure to keep the type extension in your main file, as that convention is us
 
 ### Extending the Hardhat config
 
+The boilerplate project also has an example on how to extend the Hardhat config.
+
+We strongly doing this in TypeScript and properly extending the config types.
```
Comment (fvictorio) [inline docs/advanced/building-plugins.md:79]:
```suggestion
We strongly recommend doing this in TypeScript and properly extending the config types.
```

File: docs/advanced/building-plugins.md (hunk 2 above)
Comment (alcuadrado) [reply docs/advanced/building-plugins.md:23]: This is totally outdated. Will change it now

**Comment by alcuadrado** [comment]
@fzeoli wdyt?

File: docs/advanced/building-plugins.md [hunk 5]
```diff
@@ -1,13 +1,72 @@
...
+## What exactly are plugins in Hardhat?
 
-Plugins are bits of reusable configuration. Anything that you can do in a plugin, can also be done in your config file. You can test your ideas in a config file, and move them into a plugin when ready.
+Plugins are bits of reusable configuration. Anything that you can do in a plugin, can
+also be done in your config file. You can test your ideas in a config file, and move
+them into a plugin when ready.
 
-The main things that plugins can do are extending the Hardhat Runtime Environment, extending the Hardhat config, defining new tasks, and overriding existing ones.
+When developing a plugin the main tools available to integrate new functionality are extending the
+[Hardhat Runtime Environment](../advanced/hardhat-runtime-environment), extending the Hardhat config, defining new tasks and
```
Comment (fzeoli) [inline docs/advanced/building-plugins.md:13]: declare the HRE link at the bottom and add brackets to all mentions

File: docs/advanced/building-plugins.md [hunk 6]
```diff
@@ -1,13 +1,72 @@
...
+
+Some examples of things you could achieve by creating a plugin are running a linter when
+the `check` task runs, using different compiler versions for different files or
+generating an UML diagram for your contracts.
+
+
+## Extending the Hardhat Runtime Environment
+
+Let’s go through the process of creating a plugin that adds new functionality to the Hardhat Runtime Environment. 
+By doing this, we make sure our new feature are available everywhere. That your plugin's users can access it from
```
Comment (fzeoli) [inline docs/advanced/building-plugins.md:24]:
```suggestion
By doing this, we make sure our new feature is available everywhere. This means your plugin users can access it from
```

**Pull request merged**